File: src/ast/tree/statement/continue.ts
import type { Pass } from "../../../pass/pass";
import type { Block } from "../../node/block";
import { ExecutionResult, Statement } from "../../node/statement";

export class Continue extends Statement {
    constructor(parent: Block | null) {
        super(parent);
    }

    public execute(): ExecutionResult {
        return ExecutionResult.ContinueLoop;
    }

    public visit(pass: Pass): void {
        pass.visitContinue(this);
    }
}

File: src/main/kotlin/me/honkling/miniscript/parser/ast/statement/Continue.kt
package me.honkling.miniscript.parser.ast.statement

import me.honkling.miniscript.parser.ast.Block
import me.honkling.miniscript.pass.Pass

class Continue(parent: Block) : Statement(parent) {
    override fun execute(): ExecutionResult {
        return ExecutionResult.ContinueLoop
    }

    override fun accept(pass: Pass) {
        pass.visitContinue(this)
    }
}